File: Códigos/Site 1 - Dynamic/GPT4/imagem_descrita.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import time

# Initialize the WebDriver
driver = webdriver.Chrome()

try:
    # Open the web application
    driver.get("https://the-internet.herokuapp.com/dynamic_controls")

    # Validate Page Structure
    # Check for the presence of the title and subtitle
    assert "The Internet" in driver.title
    header = driver.find_element(By.XPATH, "//h4[contains(text(), 'Dynamic Controls')]")
    assert header.is_displayed()

    # Verify the checkbox and remove button exist when the page loads
    checkbox = driver.find_element(By.XPATH, "//input[@type='checkbox']")
    remove_button = driver.find_element(By.XPATH, "//button[normalize-space()='Remove']")
    assert checkbox.is_displayed()
    assert remove_button.is_displayed()

    # Verify the text field is disabled by default
    text_field = driver.find_element(By.XPATH, "//input[@type='text']")
    assert not text_field.is_enabled()

    # Test Checkbox Removal and Addition
    # Click "Remove" and verify the checkbox disappears
    remove_button.click()
    WebDriverWait(driver, 10).until(EC.invisibility_of_element(checkbox))

    # Check if the button text changes to "Add"
    add_button = driver.find_element(By.XPATH, "//button[normalize-space()='Add']")
    assert add_button.is_displayed()

    # Click "Add" and verify the checkbox reappears
    add_button.click()
    WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.XPATH, "//input[@type='checkbox']")))
    checkbox = driver.find_element(By.XPATH, "//input[@type='checkbox']")
    assert checkbox.is_displayed()

    # Test Enable/Disable Functionality
    # Verify that the text field is disabled by default
    assert not text_field.is_enabled()

    # Click "Enable" and confirm the appearance of the message "Wait for it..."
    enable_button = driver.find_element(By.XPATH, "//button[normalize-space()='Enable']")
    enable_button.click()
    message = WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.ID, "message")))
    assert "It's enabled!" in message.text

    # Ensure the loading bar appears before the text field is enabled
    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.ID, "loading")))

    # Verify that the text field is enabled after the process is complete
    WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, "//input[@type='text']")))
    assert text_field.is_enabled()

finally:
    # Close the WebDriver
    driver.quit()

File: Códigos/Site 2 - Dynamic/Mistral/imagem_descrita.java
package site2_dynamic;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class imagem_descrita {
    private WebDriver driver;

    @BeforeMethod
    public void setup() {
        System.setProperty("webdriver.chrome.driver", "C:\\Users\\arian\\Documents\\Thinklogic\\Tools\\chromedriver-win64\\chromedriver-win64\\chromedriver.exe");
        driver = new ChromeDriver();
        driver.get("https://the-internet.herokuapp.com/dynamic_controls");
    }

    @Test(description = "Validate Page Structure")
    public void validatePageStructure() {
        WebElement title = driver.findElement(By.id("title"));
        Assert.assertTrue(title.isDisplayed(), "Title is not displayed.");

        WebElement subTitle = driver.findElement(By.cssSelector("#wirepoints h3"));
        Assert.assertTrue(subTitle.isDisplayed(), "Subtitle is not displayed.");

        WebElement checkbox = driver.findElement(By.id("checkbox"));
        Assert.assertTrue(checkbox.isDisplayed(), "Checkbox is not displayed.");

        WebElement removeButton = driver.findElement(By.id("remove-button"));
        Assert.assertTrue(removeButton.isDisplayed(), "Remove button is not displayed.");

        WebElement textField = driver.findElement(By.id("input-example"));
        Assert.assertFalse(textField.isEnabled(), "TextField should be disabled by default.");
    }

    @Test(description = "Test Checkbox Removal and Addition")
    public void testCheckboxRemovalAndAddition() {
        WebElement removeButton = driver.findElement(By.id("remove-button"));
        removeButton.click();

        // Wait for the checkbox to disappear using implicit wait
        new WebDriverWait(driver, Duration.ofSeconds(10)).until(ExpectedConditions.invisibilityOfElementLocated(By.id("checkbox")));

        WebElement addButton = driver.findElement(By.id("add-button"));
        Assert.assertEquals(addButton.getText(), "Add", "Button text should be 'Add' after removal.");
        addButton.click();

        // Wait for the checkbox to reappear using implicit wait
        new WebDriverWait(driver, Duration.ofSeconds(10)).until(ExpectedConditions.visibilityOfElementLocated(By.id("checkbox")));
    }

    @Test(description = "Test Enable/Disable Functionality")
    public void testEnableDisableFunctionality() {
        // Find the 'Enable' button and click it
        WebElement enableButton = driver.findElement(By.id("enable-button"));
        enableButton.click();

        // Wait for the message to appear using implicit wait
        new WebDriverWait(driver, Duration.ofSeconds(10)).until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[contains(text(), 'Wait for it...')]")));

        // Find the loading bar and verify its visibility
        WebElement loadingBar = driver.findElement(By.id("loading-bar"));
        Assert.assertTrue(loadingBar.isDisplayed(), "Loading bar should be displayed during the process.");

        // Wait for the text field to be enabled using implicit wait
        new WebDriverWait(driver, Duration.ofSeconds(10)).until(ExpectedConditions.elementToBeClickable(driver.findElement(By.id("input-example"))));

        // Find the text field and verify its enabling
        WebElement textField = driver.findElement(By.id("input-example"));
        Assert.assertTrue(textField.isEnabled(), "TextField should be enabled after the process is complete.");
    }

    @AfterMethod
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}
